atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -479,22 +479,16 @@ public CassandraServer getRandomCassandraNodeForKey(byte[] key) {
         return getRandomGoodHost().getCassandraServer();
     }
 
-    public void addPool(CassandraServer server) {
+    public CassandraClientPoolingContainer createPool(CassandraServer server) {
         int currentPoolNumber = cassandraHosts.indexOf(server) + 1;
-        addPoolInternal(
-                server,
-                new CassandraClientPoolingContainer(metricsManager, server, config, currentPoolNumber, poolMetrics));
+        return new CassandraClientPoolingContainer(metricsManager, server, config, currentPoolNumber, poolMetrics);
     }
 
-    public void returnOrCreatePool(CassandraServer server, Optional<CassandraClientPoolingContainer> container) {
```

Comment:
My understanding is that users do not directly use this class.. therefore should not be an API break?